src/SpinModelRegion.cpp

```diff
@@ -80,12 +85,14 @@ namespace geopm
             ModelRegion::region_enter();
             for (uint64_t i = 0 ; i < m_num_progress_updates; ++i) {
                 ModelRegion::loop_enter(i);
-
                 double timeout = 0.0;
                 struct geopm_time_s start = {{0,0}};
                 struct geopm_time_s curr = {{0,0}};
                 (void)geopm_time(&start);
                 while (timeout < m_delay) {
+                    if (m_run_atom) {
```

Comment:
```m_run_atom``` is not initialized in the SpinModelRegion constructor.  I'm wondering if it would be more safe to explicitly set it to ```NULL``` there and then make this check if it is still ```NULL```.

Reply:
It's not a function pointer, its a std::function.  It has bool method:

https://en.cppreference.com/w/cpp/utility/functional/function/operator_bool

Reply:
Fancy.  I didn't realize it worked that way.